replica/table.cc

```diff
@@ -2517,6 +2517,24 @@ table::make_partition_presence_checker(lw_shared_ptr<const sstables::sstable_set
     };
 }
 
+max_purgeable_fn table::get_max_purgeable_fn_for_compacting_reader() {
+    return [this] (const dht::decorated_key& dk, ::is_shadowable is_shadowable) {
+        auto max_purgeable_timestamp = api::max_timestamp;
+
```

Comment:
we can easily grab the storage group that overlaps with dk.token(). see storage_group_for_token(), so we reduce the search space.